scripts/get-plantuml-jar.js

```diff
@@ -29,19 +38,58 @@ function download (uri, filename) {
       // Success, pipe to file
       var fileStream = fs.createWriteStream(filename)
       res.pipe(fileStream)
+      if (callback) {
+        res.on('end', function () {
+          callback()
+        })
+      }
     } else if (res.headers.location) {
       // Follow redirect
-      download(res.headers.location, filename)
+      download(res.headers.location, filename, callback)
     } else {
       // Error
       trace.forEach(function (line) {
         console.error(line)
       })
-      console.error('Failed to download plantuml.jar')
+      var error = 'Failed to download ' + filename
+      console.error(error)
       process.exitCode = 1
+      if (callback) {
+        callback(error)
+      }
     }
   })
 }
 
 console.log('Downloading plantuml.jar version ' + plantumlVersion)
-download(PLANTUML_FILES_URL + PLANTUML_FILES_JAR_PATH, PLANTUML_JAR)
+download(PLANTUML_FILES_URL + PLANTUML_FILES_JAR_PATH, PLANTUML_JAR, function (err) {
+  if (!err) {
+    // once downloaded, check if dot is installed using plantuml's testdot command
+    var plantuml = require('../lib/node-plantuml')
```

Comment:
Could you explain how we know node-plantuml is always available in this location?

Reply:
Given `require(x)`, where x is a relative path, x will always be relative to the directory of the containing module.